scala/project/ProjectPlugin.scala

```diff
@@ -0,0 +1,40 @@
+import sbt._
+import sbt.Keys._
+
+object ProjectKeys {
+  def KindProjector =
+    addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.4")
+}
+
+object ProjectPlugin extends AutoPlugin {
```

Comment:
sorry, your sbt layout was causing me pain. I'm utterly intolerable, I know.

Reply:
You mentioned in gitter that the `commonSetting` way is broken. How come? I'd like to know, since all our Scala projects at work do that.

Reply:
commonSettings does not apply to the autogen let project.

Reply:
I could've sworn it did.